src/QueryDescription/index.js

```diff
@@ -253,6 +313,7 @@ export function queryWithoutDeleted(query: QueryDescription): QueryDescription {
   const { join, where: whereConditions } = query
 
   const newQuery = {
+    ...query,
     join: [...join, ...joinsWithoutDeleted(join)],
     where: [...whereConditions, whereNotDeleted],
   }
```

Comment:
i think you also have to fix up `Query.extend()` -- i guess it would be better if it called a function defined in _this_ file, and not manipulated on `join, where` directly to avoid mistakes in the future